traffic_monitor/todata/todata.go

```diff
@@ -179,6 +187,10 @@ func (d TODataThreadsafe) Update(to towrap.TrafficOpsSessionThreadsafe, cdn stri
 		return fmt.Errorf("getting server types from monitoring config: %v", err)
 	}
 
+	if val, ok := mc.Config["tm.sameipservers.control"]; ok && val.(string) == "true" {
+		newTOData.SameIpServers = getSameIPServers(mc)
+	}
```

Comment:
If the condition is false, `newTOData.SameIpServers` will be `nil`, so should be an `else` here creating an empty map.